app/src/main/java/com/kodeco/android/countryinfo/MainActivity.kt

```diff
@@ -3,21 +3,39 @@ package com.kodeco.android.countryinfo
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
-import com.kodeco.android.countryinfo.ui.components.CountryInfoScreen
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.Surface
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import com.kodeco.android.countryinfo.network.CountryUIState
+import com.kodeco.android.countryinfo.network.RetrofitClient
+import com.kodeco.android.countryinfo.ui.components.MainScreen
 import com.kodeco.android.countryinfo.ui.theme.MyApplicationTheme
+import kotlinx.coroutines.delay
 
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-
-        // TODO build out the retrofit service and reference it here.
-        //  Pass the service in to the CountryInfoScreen composable below.
-
         setContent {
             MyApplicationTheme {
-                // TODO complete the composable content and provide
-                //  models for Country, CountryName, and CountryFlags.
-                CountryInfoScreen()
+                var uiState: CountryUIState by rememberSaveable { mutableStateOf(CountryUIState.Loading) }
+                Surface(
+                    modifier = Modifier.fillMaxSize()
+                ) {
+                    MainScreen(uiState)
+                    LaunchedEffect(Unit) {
+                        delay(3000)
```

Comment:
this delay is not needed

Reply:
Thank you for the review @abunur 
I will remove delay and merge the PR